### ISSUE TITLE
 Error when creating relationship field with properties using `update` field on `update`

### ISSUE THREAD

**Issue by dmoree** [neo4j/graphql#337]
**Describe the bug**
The error ``Variable `this_X0_create0_relationship` not defined`` is thrown while using the `updateY` mutation and creating a node through a `@relationship` field **only** if the relationship has properties. The node is created if relationship properties are not defined. The relationship is not being assigned a variable in the translation.

**Type definitions**
Based on the `neo-push` example.

**To Reproduce**

1. Clone repository
```
git clone https://github.com/dmoree/neo4j-graphql-issue-create-relationship.git
```
2. Seed and start server
```
cd neo4j-graphql-issue-create-relationship && yarn install && yarn seed && yarn start
```
3. Run the following `mutation` in GraphiQL at `http://localhost:4000/graphql` (or as defined in `.env`)
```graphql
mutation UpdateBlogUpdate {
  updateBlogs(
    where: { name: "blog" }
    update: {
      posts: {
        create: {
          node: {
            title: "title"
            content: "content"
            author: {
              connect: { where: { node: { email: "admin@admin.com" } } }
            }
          },
          relationship: {
            property: "property"
          }
        }
      }
    }
  ) {
    blogs {
      id
      postsConnection {
        totalCount
        edges {
          node {
            id
            title
            content
          }
          property
        }
      }
    }
  }
}
```
4. See error:
```
  @neo4j/graphql:execute Neo4jError: Variable `this_posts0_create0_relationship` not defined (line 17, column 5 (offset: 803))
"SET this_posts0_create0_relationship.property = $this_posts0_create0_relationship_property"
     ^
```

**Expected behavior**
Expected a post to be created with appropriate relationship property.

**System (please complete the following information):**

-   OS: macOS 11.4
-   Version: `@neo4j/graphql@2.0.0-alpha.3`
-   Node.js version: `14.16.0`

**Additional context**
Debugger output:
```
  @neo4j/graphql:execute About to execute Cypher:
Cypher:
MATCH (this:Blog)
WHERE this.name = $this_name

WITH this
CREATE (this_posts0_create0_node:Post)
SET this_posts0_create0_node.createdAt = datetime()
SET this_posts0_create0_node.id = randomUUID()
SET this_posts0_create0_node.title = $this_posts0_create0_node_title
SET this_posts0_create0_node.content = $this_posts0_create0_node_content
WITH this, this_posts0_create0_node
OPTIONAL MATCH (this_posts0_create0_node_author_connect0_node:User)
WHERE this_posts0_create0_node_author_connect0_node.email = $this_posts0_create0_node_author_connect0_node_email
FOREACH(_ IN CASE this_posts0_create0_node_author_connect0_node WHEN NULL THEN [] ELSE [1] END | 
MERGE (this_posts0_create0_node)<-[:WROTE]-(this_posts0_create0_node_author_connect0_node)
)
MERGE (this)-[:HAS_POST]->(this_posts0_create0_node)
SET this_posts0_create0_relationship.property = $this_posts0_create0_relationship_property

WITH this
CALL {
WITH this
MATCH (this)-[this_has_post:HAS_POST]->(this_post:Post)
WITH collect({ property: this_has_post.property, node: { id: this_post.id, title: this_post.title, content: this_post.content } }) AS edges
RETURN { edges: edges, totalCount: size(edges) } AS postsConnection
}
RETURN this { .id, postsConnection } AS this
Params:
{
  "this_name": "blog",
  "this_posts0_create0_node_title": "title",
  "this_posts0_create0_node_content": "content",
  "this_posts0_create0_node_author_connect0_node_email": "admin@admin.com",
  "this_posts0_create0_relationship_property": "property"
} +26m
  @neo4j/graphql:execute Neo4jError: Variable `this_posts0_create0_relationship` not defined (line 17, column 5 (offset: 803))
"SET this_posts0_create0_relationship.property = $this_posts0_create0_relationship_property"
     ^
: 
    at captureStacktrace (/Users/dmoree/projects/neo4j/graphql/neo4j-graphql-issue-create-relationship/node_modules/neo4j-driver-core/lib/result.js:239:17)
    at new Result (/Users/dmoree/projects/neo4j/graphql/neo4j-graphql-issue-create-relationship/node_modules/neo4j-driver-core/lib/result.js:59:23)
    at newCompletedResult (/Users/dmoree/projects/neo4j/graphql/neo4j-graphql-issue-create-relationship/node_modules/neo4j-driver-core/lib/transaction.js:372:12)
    at Object.run (/Users/dmoree/projects/neo4j/graphql/neo4j-graphql-issue-create-relationship/node_modules/neo4j-driver-core/lib/transaction.js:226:20)
    at Transaction.run (/Users/dmoree/projects/neo4j/graphql/neo4j-graphql-issue-create-relationship/node_modules/neo4j-driver-core/lib/transaction.js:98:34)
    at /Users/dmoree/projects/neo4j/graphql/neo4j-graphql-issue-create-relationship/node_modules/@neo4j/graphql/src/utils/execute.ts:84:16
    at TransactionExecutor._safeExecuteTransactionWork (/Users/dmoree/projects/neo4j/graphql/neo4j-graphql-issue-create-relationship/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:92:26)
    at TransactionExecutor._executeTransactionInsidePromise (/Users/dmoree/projects/neo4j/graphql/neo4j-graphql-issue-create-relationship/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:83:34)
    at /Users/dmoree/projects/neo4j/graphql/neo4j-graphql-issue-create-relationship/node_modules/neo4j-driver-core/lib/internal/transaction-executor.js:40:19
    at new Promise (<anonymous>) {
  constructor: [Function: Neo4jError],
  code: 'Neo.ClientError.Statement.SyntaxError'
} +68ms
```
It appears as if the generated cypher query is good, but the relationship `:HAS_POST` is not being assigned the expected variable `this_posts0_create0_relationship`. _i.e._
```
MERGE (this)-[:HAS_POST]->(this_posts0_create0_node)
```
should have been translated as:
```
MERGE (this)-[this_posts0_create0_relationship:HAS_POST]->(this_posts0_create0_node)
```

**Comment by darrellwarde** [comment]
Many thanks for the detailed error report @dmoree, exactly the kind of feedback we're looking for on the alpha! We'll take a look. 🙂

**Comment by darrellwarde** [comment]
This will be released in 2.0.0-alpha.4 very shortly! 🙂
